src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -180,7 +180,10 @@ public void setDefinition(FlowDefinition definition) {
     @Override protected void submit(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, Descriptor.FormException {
         super.submit(req, rsp);
         JSONObject json = req.getSubmittedForm();
-        definition = req.bindJSON(FlowDefinition.class, json.getJSONObject("definition"));
+        // TODO using req.bindJSON(FlowDefinition.class, json.getJSONObject("definition")) breaks CpsFlowDefinitionTest
+        definition = json.get("definition") == null ? null :
+                Descriptor.newInstancesFromHeteroList(req, json.getJSONObject("definition"),
+                        ExtensionList.lookup(FlowDefinitionDescriptor.class)).get(0);
```

Comment:
Either re-add the comment from https://github.com/jenkinsci/workflow-job-plugin/commit/a25d15548158461749870d328746e2be482f9e4d about switching to `Descriptor.bindJSON` in the future, or do it now since the baseline is already high enough (untested):
```suggestion
        definition = Descriptor.bindJSON(req, FlowDefinition.class, json.get("definition"));
```